src/vs/gitpod/browser/gitpodInsightsAppender.ts

```diff
@@ -6,14 +6,17 @@
 
 import { IProductService } from 'vs/platform/product/common/productService';
 import { ITelemetryAppender } from 'vs/platform/telemetry/common/telemetryUtils';
-import { mapTelemetryData, SenderKind } from 'vs/gitpod/common/insightsHelper';
+import { mapMetrics, mapTelemetryData } from 'vs/gitpod/common/insightsHelper';
+import type { IDEMetric } from '@gitpod/ide-metrics-api-grpcweb';
+
+type SendMetrics = (metrics: IDEMetric[]) => Promise<void>;
 
 export class GitpodInsightsAppender implements ITelemetryAppender {
-	private _baseProperties: { appName: string; uiKind: 'web'; version: string };
+	private readonly _baseProperties: { appName: string; uiKind: 'web'; version: string };
+	private readonly devMode = this.productService.nameShort.endsWith(' Dev');
```

Comment:
This doesn't look nice, should be moved to workbench.ts and if true then set `gitpodPreview` to `undefined` and use that as a check here